articles/implementing-shooting-method-in-matlab/index.md

```diff
@@ -0,0 +1,191 @@
+
+![Question](/engineering-education/Implementing-shooting-method-in-Matlab/shooting_three.png)
+
+Where;
+EI=flexual rigidity
+w = Magnitude of the equally distributed load
+l=length
+We will look at how to solve this problem using Matlab.
+We first define our variables.
+
+```Matlab
+%Shooting method example for linear BVP
+E = 200*10^9;      %Pa
+I = 30000*(1/100)^4;     %m4
+w = 15000;             %N/m
+L = 3;               %m
+```
+
+This is followed by definning the linear differential equation
+
+```matlab
+yp = @(x,y) [y(2); 1/(E*I)*((w*L*x/2)-(w*x^2)/2)];
+```
+
+The above is our $\frac{dy}{dx}$ function. Just like we have been using our IVP, we've written the differential equation as a system of two. The systems are first-order differential equation and the $\frac{dy}{dx}$ function. It returns a column vector of
+
+$$
+[dy_1 dx\\dy_2 dx]
+$$
+
+Now give the boundary conditions
+
+```matlab
+y0 = 0; yL = 0;
+```
+
+Then we have our guesses at the initial condition. Since this is a linear differential equation, we need two guesses. We now give our guesses.
+
+```matlab
+ICguesses = [0 -1];
+```
+
+`ICguesses` is like our aim, and we will aim at the IVP solver, which is ODE45, at that target boundary condition using the two guesses. So far, our first integration we call ODE45 with the first guesses.
+
+```matlab 
+[x1, y1] = ode45(yp, [0 L], [y0 ICguesses(1)]);
+```
+
+Note that from our first guess, always save the last value of `y1` in the first integration since this is the value we are interested in. Furthermore, we want to compare this last value of `y1` to the target boundary condition `y2`. Therefore, it is saved as `shot1`.
+```Matlab
+shot1 = y1(end,1);
+```
+
+Now conduct the second 'shot' using the `ICguesses(2)` and save the last value as `shot2`.
+
+```matlab
+[x2, y2] = ode45(yp, [0 L], [y0 ICguesses(2)]);
+shot2 = y2(end,1);
+```
+
+Then put the 'shots' together in a vector.
+
+```Matlab
+shots = [shot1 shot2];
+```
+
+The next step is to perform the interpolation.
+
+```matlab
+newIC = interp1(shots, ICguesses, yL, 'linear', 'extrap');
+```
+
+To demonstrate what is happening in the code above, we look at the sketch below. We have our `ICguesses` on the `y-axis` and `y-end` on the `x-axis`.
+
+![demonstration](shooting_four.png)
+
+Now, that's the plot for the 'shots' and the guesses. We then want to know the guess to get `yl`. So we have three points that we know for the interpolation. They are all our target boundary conditions, that is, shots to the boundary condition and the actual target boundary condition `$y_l$`. The unknown value is the unknown guess. It is why we use them for the call to the enterprise. Shots are the `x-values`, and `ICguess` the y-values. Since this is a linear equation, we use the linear interpolation defined by `linear interp`. The last call in that function is the `extrap`, which means extrapolation. It is in case the `$y_l$` is not between the two shots.
```

Comment:
Kindly break this paragraph down, it's too long. ( you can also leave me comment if it's not possible and why.)